### PR TITLE
Feature/22136 widget from dataform

### PR DESCRIPTION
**Pull request by ozelenyi**
Add widget that's configurable through data form

### CODE REVIEW

File: src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        return this.spec.types.filter((t) => t.key === vTypeKey).shift();
+    }
+
+    public getEntityField(field) {
+        if (this.xmEntity) {
+            return this.xmEntity.data[field] || '';
+        }
+    }
+
+    // public onEdit(entity: XmEntity, entitySpec: XmEntitySpec) {
```
Comment (Kaskyi) [inline src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts]: pls remove

File: src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                        this.jsfAttributes.data = Object.assign(
+                            nullSafe(this.jsfAttributes.data),
+                            nullSafe(this.xmEntity.data));
+                    }
+                }),
+            ).subscribe();
+
+    }
+
+    // private processForms(dataForm) {
```
Comment (Kaskyi) [inline src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts]: pls remove

File: src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    styleUrls: ['./entity-card-widget.component.scss']
+})
+export class EntityCardWidgetComponent implements OnInit {
+
+    public config: any;
+    public xmEntity: XmEntity;
+    public xmEntitySpec: XmEntitySpec;
+    public spec: Spec;
+
+    // isEdit = true;
```
Comment (Kaskyi) [inline src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts]: pls remove

File: src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts (hunk 1 above)
Comment (ozelenyi) [reply src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts]: It`s needed for future rework

File: src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts (hunk 2 above)
Comment (ozelenyi) [reply src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts]: It`s needed for future rework

File: src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts (hunk 3 above)
Comment (ozelenyi) [reply src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts]: It`s needed for future rework

File: src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts (hunk 1 above)
Comment (Kaskyi) [reply src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts]: can u implement it in this pr?

Comment (ozelenyi) [reply src/app/ext-commons/ext-common-entity/entity-card-widget/entity-card-widget.component.ts]: Nope

**Pull request closed without merging**